File: corporate-reporting/src/types/PartnerOrganisationRiskMitigationAction.ts
import { IUser } from "./User";
import { IPartnerOrganisationRisk } from "./PartnerOrganisationRisk";
import { IReportingEntityWithDeliveryDates, ReportingEntityWithDeliveryDates } from "./ReportingEntity";
import { IPartnerOrganisationRiskMitigationActionUpdate } from "./PartnerOrganisationRiskMitigationActionUpdate";

export interface IPartnerOrganisationRiskMitigationAction extends IReportingEntityWithDeliveryDates {
    Description: string;
    PartnerOrganisationRiskID: number;
    RiskMitigationActionCode: number;
    OwnerUserID: number;
    ActionIsOngoing: boolean;
    PartnerOrganisationRisk?: IPartnerOrganisationRisk;
    OwnerUser?: IUser;
    PartnerOrganisationRiskMitigationActionUpdates?: IPartnerOrganisationRiskMitigationActionUpdate[];
}

export class PartnerOrganisationRiskMitigationAction extends ReportingEntityWithDeliveryDates implements IPartnerOrganisationRiskMitigationAction {
    public Description = '';
    public PartnerOrganisationRiskID = null;
    public RiskMitigationActionCode = null;
    public OwnerUserID = null;
    public ActionIsOngoing = false;
    public Contributors = [];
}

File: ORB.Core/Models/PartnerOrganisationRiskMitigationAction.cs
using System;
using System.Collections.Generic;

namespace ORB.Core.Models
{
    public partial class PartnerOrganisationRiskMitigationAction : ReportingSubEntity
    {
        public PartnerOrganisationRiskMitigationAction()
        {
            PartnerOrganisationRiskMitigationActionUpdates = new HashSet<PartnerOrganisationRiskMitigationActionUpdate>();
        }

        public int PartnerOrganisationRiskID { get; set; }
        public int? RiskMitigationActionCode { get; set; }
        public DateTime? BaselineDate { get; set; }
        public DateTime? ForecastDate { get; set; }
        public DateTime? ActualDate { get; set; }
        public int? OwnerUserID { get; set; }
        public bool? ActionIsOngoing { get; set; }

        public virtual User OwnerUser { get; set; }
        public virtual PartnerOrganisationRisk PartnerOrganisationRisk { get; set; }
        public virtual ICollection<PartnerOrganisationRiskMitigationActionUpdate> PartnerOrganisationRiskMitigationActionUpdates { get; set; }
    }
}
